packages/x-charts/src/context/CartesianProvider/computeValue.ts

```diff
@@ -0,0 +1,182 @@
+import { scaleBand, scalePoint } from 'd3-scale';
+import { DEFAULT_X_AXIS_KEY, DEFAULT_Y_AXIS_KEY } from '../../constants';
+import { AxisConfig, ChartsXAxisProps, ScaleName } from '../../models';
+import { ChartsYAxisProps, isBandScaleConfig, isPointScaleConfig } from '../../models/axis';
+import { CartesianChartSeriesType, ExtremumGetter } from '../../models/seriesType/config';
+import { DefaultizedAxisConfig } from './CartesianContext';
+import { getColorScale, getOrdinalColorScale } from '../../internals/colorScale';
+import { getTickNumber } from '../../hooks/useTicks';
+import { getScale } from '../../internals/getScale';
+import { DrawingArea } from '../DrawingProvider';
+import { FormattedSeries } from '../SeriesContextProvider';
+import { MakeOptional } from '../../models/helpers';
+import { getAxisExtremum } from './getAxisExtremum';
+
+const DEFAULT_CATEGORY_GAP_RATIO = 0.2;
+const DEFAULT_BAR_GAP_RATIO = 0.1;
+
+export const computeValue = (
+  drawingArea: DrawingArea,
+  formattedSeries: FormattedSeries,
+  xAxis: MakeOptional<AxisConfig<ScaleName, any, ChartsXAxisProps>, 'id'>[] | undefined,
+  yAxis: MakeOptional<AxisConfig<ScaleName, any, ChartsYAxisProps>, 'id'>[] | undefined,
+  xExtremumGetters: { [K in CartesianChartSeriesType]?: ExtremumGetter<K> },
+  yExtremumGetters: { [K in CartesianChartSeriesType]?: ExtremumGetter<K> },
+) => {
```

Comment:
Is there something preventing us to refactor a bit deeper with 

```ts
function computeValue(
  drawingArea: DrawingArea,
  formattedSeries: FormattedSeries,
  axis: MakeOptional<AxisConfig<ScaleName, any, ChartsYAxisProps>, 'id'>[] | undefined,
  extremumGetters: { [K in CartesianChartSeriesType]?: ExtremumGetter<K> },
)
function computeValue(
  drawingArea: DrawingArea,
  formattedSeries: FormattedSeries,
  axis: MakeOptional<AxisConfig<ScaleName, any, ChartsXAxisProps>, 'id'>[] | undefined,
  extremumGetters: { [K in CartesianChartSeriesType]?: ExtremumGetter<K> },
) {
	// ...
	return { completedAxis, axisIds }
}
```

Reused with 

```ts
const value = React.useMemo(
    () => {
	const { completedAxis: completedXAxis, axisIds: xAxisIds } = computeValue(drawingArea, formattedSeries, xAxis, xExtremumGetters);
	const { completedAxis: completedYAxis, axisIds: yAxisIds } = computeValue(drawingArea, formattedSeries, yAxis, yExtremumGetters);
	return {
    	isInitialized: true,
   	 data: {
     	 xAxis: completedXAxis,
     	 yAxis: completedYAxis,
     	 xAxisIds,
     	 yAxisIds,
    },
}
```

Reply:
I think it is feasible, there are only small changes between x/y which could be changed with a flag.

Reply:
updated